cli/cmd/vuln_host_show_assessment.go

```diff
@@ -58,6 +58,22 @@ Grab a CVE id and feed it to the command:
 					vulCmdState.Details = true
 				}
 			}
+
+			// validate collector_type flag
+			if vulCmdState.CollectorType != vulnHostCollectorTypeAgentless && vulCmdState.CollectorType != vulnHostCollectorTypeAgent {
+				return errors.Errorf("collector_type must be either %s or %s", vulnHostCollectorTypeAgent, vulnHostCollectorTypeAgentless)
+			}
+
+			if vulCmdState.CollectorType == vulnHostCollectorTypeAgentless {
+				if !checkAgentlessCloudAccount() {
+					if cmd.Flags().Changed("collector_type") {
```

Comment:
This logic tricked me for a few.

We will throw an error message only, and only if
1. The user does NOT have an Agentless cloud-account integration (AND)
2. The user used the flag `--collector_type` (AND)
3. ONLY if the provided collector type is `Agentless`, which is already the default 😕, which then I thought the flag never changed..... 😖  

We might wanna find a better way to do this check.

Finally, we are already interacting with the APIs here so, maybe displaying a Spinner about the task that the CLI is doing, would be nice for folks with slow internet.

Reply:
An explanation of the behaviour:

If the user explicitly wants agentless results eg. they've set the flag. But they've not configured an agentless integration then display an error rather than agent results. 

If no flag is set, and there is no agentless integration then we just default to agent

Reply:
I'll add comments around this to make it a little more clear what the logic is.